.github/workflows/prerelease.yml

```diff
@@ -1,69 +1,71 @@
+      - "master"
 
 jobs:
   prerelease:
-    if: "github.event_name == 'workflow_dispatch' || contains(github.event.head_commit.message, '[prerelease]')"
+    if: "contains(github.event.head_commit.message, '[prerelease]')"
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
         with:
           fetch-depth: 0
-      - uses: ./.github/actions/setup
+      - name: Run Setup
+        uses: ./.github/actions/setup
         with:
-          npm_token: ${{secrets.npm_token}}
+          npm_token: ${{ secrets.npm_token }}
       - uses: ./.github/actions/git-creds
       - name: Check if there are changed packages
-        id: changed-packages
         run: |
           changed_packages=$(yarn -s lerna ls --since=origin/master --json --loglevel=error)
-          [[ $changed_packages = "[]" ]] && echo "No packages changed in this branch" && exit 1 || echo "Continuing to prerelease"
+          [[ $changed_packages = "[]" ]] && echo "Skipping prerelease. Nothing to prerelease as Lerna didn't detect any changes." && exit 1 || echo "Changed detected. Continuing to prerelease" && echo "$changed_packages"
       - name: Build all affected packages
-        run: yarn lerna run build --since=origin/master
-      - name: Generate prerelease prefix
-        run: |
-          normalized_branch_name=$(git rev-parse --abbrev-ref HEAD | tr -dc '[:alnum:]\n\r' | tr '[:upper:]' '[:lower:]')
-          echo "BRANCH=$normalized_branch_name" >> $GITHUB_ENV
+        run: yarn lerna run build --since=origin/master --include-dependencies
+      - name: Run Tests
+        shell: bash
+        run: yarn lerna run test --since=origin/master
       - name: Generate new versions
-        run: yarn lerna version --conventional-commits --conventional-prerelease --no-changelog --no-push --preid $BRANCH -y
+        id: generate-versions
+        run: |
+          preid=${{ github.ref_name }}
+          normalized_pre_id=$(echo "$preid" | tr -dc '[:alnum:]\n\r' | tr '[:upper:]' '[:lower:]')
+
+          full_output=$(yarn lerna version --conventional-commits --conventional-prerelease --json --no-changelog --no-push --preid "$normalized_pre_id" -y)
+
+          # extract only the JSON portion from the full output, excluding yarn logs
+          json_output=$(echo "$full_output" | awk '/^\[/{p=1} p; /\]/{p=0}' )
+
+          echo 'new_versions<<EOF' >> $GITHUB_OUTPUT
+          echo "$json_output" >> $GITHUB_OUTPUT
+          echo 'EOF' >> $GITHUB_OUTPUT
       - run: yarn config set registry https://registry.npmjs.org/
       - name: Setup .npmrc for publish
         run: npm set "//registry.npmjs.org/:_authToken=$NODE_AUTH_TOKEN"
         env:
-          NODE_AUTH_TOKEN: ${{secrets.npm_token}}
-      - run: npm whoami
+          NODE_AUTH_TOKEN: ${{ secrets.npm_token }}
       - name: Publish new versions
-        run: yarn lerna publish from-package --dist-tag $BRANCH -y
+        run: yarn lerna publish from-package --dist-tag prerelease -y
```

Comment:
This means that if several people try to prerelease at the same time they will override each other. This can be very very unexpected and frustrating, not sure why it's needed.

Reply:
We wanted to avoid a case where we have lots of random tags along with "latest", "version1" etc. We thought it might be confusing for consumers to know which version to take, it might be chaotic.
Maybe we should instead prefix the tag with `prerelease-` or something? WDYT?
<img width="854" alt="Screenshot 2024-02-18 at 13 35 52" src="https://github.com/mondaycom/monday-ui-react-core/assets/18269880/679e0942-9794-4bc4-baf4-35b988b04185">

Reply:
Sure, sounds good!
Generally speaking, I don't think people really bother with tags - they just install and get the default `latest`.